packages/cardano-services-client/src/HttpProvider.ts

```diff
@@ -78,11 +78,9 @@ export const createHttpProvider = <T extends Provider>({
   new Proxy<T>({} as T, {
     // eslint-disable-next-line sonarjs/cognitive-complexity
     get(_, prop) {
-      if (prop === 'then') return;
       const method = prop as keyof T;
       const urlPath = paths[method];
-      if (!urlPath)
-        throw new ProviderError(ProviderFailure.NotImplemented, `HttpProvider missing path for '${prop.toString()}'`);
+      if (!urlPath) return;
```

Comment:
We don't want to remove a function, because we want this behavior
```ts
let obj = {};
typeof obj.undefinedProp === 'undefined'
// instead of
typeof obj.undefinedProp === 'function'
```